File: q1935/q1935.py
class Solution:
    def canBeTypedWords(self, text: str, brokenLetters: str) -> int:
        brokenLetterSet = set(brokenLetters)

        res = 0
        for word in text.split(" "):
            if self.canBeTyped(word, brokenLetterSet):
                res += 1
        return res

    
    def canBeTyped(self, word: str, brokenLetterSet: set) -> bool: 
        for c in word: 
            if c in brokenLetterSet: 
                return False
        return True
        

File: q1935/q1935.java
class Solution {
    public int canBeTypedWords(String text, String brokenLetters) {
        Set<Character> brokenLettersSet = new HashSet<>();
        for (char c : brokenLetters.toCharArray()) brokenLettersSet.add(c);

        int res = 0;
        for (String word : text.split(" ")) {
            if (canBeTyped(word, brokenLettersSet)) {
                res++;
            }
        }
        return res;
    }

    private boolean canBeTyped(String word, Set<Character> brokenLettersSet) {
        for (char c : word.toCharArray()) {
            if (brokenLettersSet.contains(c)) return false;
        } 
        return true;
    }
}
